src/server/components/lists/controllers/ingest/__tests__/helpers.test.ts

```diff
@@ -0,0 +1,69 @@
+import { arrayHasChanges, getObjectDiff } from "../helpers";
+
+test("arrayHasChanges", () => {
```

Comment:
do you need to add a test for booleans as well? I think there are some flags providers can set

Reply:
Yep. I've changed the algo slightly actually. Issue before was that it was checking if the value existed at all in the array, but it should've been checking by index.

e.g. 

[true, true, false] and [true, false, false] would've returned false (no changes) because true/false exist at least once (findIndex) in both arrays. But what should be tested is "there are 2 trues, and 1 false - does the after array contain 2 trues and 1 false)".

I've done a default sort `beforeArray.sort()` (I don't know why ts/eslint actually complains about this. the default js sort is fine, as long as it's consistent for both), then iterate to check if beforeArray at position 1 === afterArray at position 1